pkg/pool-weighted/contracts/BaseWeightedPool.sol

```diff
@@ -82,6 +82,14 @@ abstract contract BaseWeightedPool is BaseMinimalSwapInfoPool {
 
     /**
      * @dev Returns the current value of the invariant.
+     *
+     * **IMPORTANT NOTE**: calling this function within a Vault context (i.e. in the middle of a join or an exit) is
```

Comment:
We should also tell users _how_ to protect themselves, otherwise we're giving a warning but no actionable advice.

Reply:
how about now? https://github.com/balancer-labs/balancer-v2-monorepo/pull/2205/commits/ab7702eae6f4b17e50cdb89f1e52923e9d6d674e